tests/test_models.py

```diff
@@ -546,6 +546,118 @@ def test_multi_gpu_model_ddp():
     run_gpu_model_test(trainer_options, model, hparams)
 
 
+def test_model_checkpoint_options():
+    """
+    Test ModelCheckpoint options
+    :return:
+    """
+    def my_own_save_function(filepath):
+        open(filepath, 'a').close()
```

Comment:
How about we change name to mock_save_function, it will communicate better the intent.
Also it would be better to not save things to disk, instead the function could be updating a dictionary ( this would require mocking the file deletion function as well ).

Reply:
Hi @VismantasD I agree with you that during the test this could be much better if we mock the functions instead of saving things to disk.
But it seems that in the `testing_utils.py`, these functions are implemented with doing real operations on the disk.
![image](https://user-images.githubusercontent.com/10709657/68101976-80139180-ff0b-11e9-85b5-a07c9007b9c5.png)

I feel that this is a much bigger change beyond this pull request, and I think it might be bettter if we open a new issue for this.